### PR TITLE
Make the checks non-strict again

### PR DESCRIPTION
**Pull request by leofeyer**
This is how I would adjust https://github.com/contao/contao/pull/3391.

### CODE REVIEW

File: core-bundle/src/Resources/contao/library/Contao/Database.php [hunk 1]
```diff
@@ -551,7 +551,7 @@ public function getChildRecords($arrParentIds, $strTable, $blnSorting=false, $ar
 			}
 		}
 
-		return $arrReturn;
+		return array_map('\intval', $arrReturn);
```
Comment (Toflar) [inline core-bundle/src/Resources/contao/library/Contao/Database.php:554]: That change looks weird?

Comment (leofeyer) [reply core-bundle/src/Resources/contao/library/Contao/Database.php:554]: It is the same thing you have added in the `getParentRecords()` method on line 574. 😊 Just making sure the method is returning integers, too.

Comment (Toflar) [reply core-bundle/src/Resources/contao/library/Contao/Database.php:554]: But this PR is all about reversing that change? :D

Comment (leofeyer) [reply core-bundle/src/Resources/contao/library/Contao/Database.php:554]: It is about removing the strict checks in the DC_Table class. Returning integers in the Database methods sounds like a good idea to me, though.

**Pull request merged**